File: cpp/581_最短无序连续子数组_medium.cpp
class Solution {
public:
    int findUnsortedSubarray(vector<int>& nums) {
        int left = -1, right = -1;
        int minn = INT_MAX;
        int maxn = INT_MIN;
        int n = nums.size();
        for (int i=0; i<n; i++) {
            if (maxn > nums[i])
                right = i;
            else
                maxn = nums[i];
            if (minn < nums[n-i-1])
                left = n-i-1;
            else
                minn = nums[n-i-1];
        }
        return right == -1 ? 0 : right-left+1;

    }
};

File: python/581_最短无序连续子数组_medium.py
class Solution:
    def findUnsortedSubarray(self, nums: List[int]) -> int:
        left, minn = -1, float('inf')
        right, maxn = -1, float('-inf')
        n = len(nums)
        for i in range(n):
            if maxn > nums[i]:
                right = i
            else:
                maxn = nums[i]
            if minn < nums[n-i-1]:
                left = n-i-1
            else:
                minn = nums[n-i-1]
        return 0 if right == -1 else right - left + 1